include/device.h

```diff
@@ -41,15 +74,20 @@ struct device {
   /* Device information and state. */
   device_bus_t bus;
   driver_t *driver;
+  // TODO: most likeley we want to get rid of `instance` field
   void *instance; /* used by bus driver to store data in children */
   void *state;    /* memory requested by driver for its state*/
+  device_attr_t attributes[DEV_ATTR_MAX]; /* equivalent of FreeBSD's `ivars` */
```

Comment:
Now that comment is not exactly true and I think it's more confusing than helpful.